independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java

```diff
@@ -694,7 +695,19 @@ private static Set<InjectableBean<?>> resolve(List<InjectableBean<?>> matching)
         nonDefault.removeIf(InjectableBean::isDefaultBean);
         if (nonDefault.isEmpty()) {
             // All the matching beans were default
-            return Set.copyOf(matching);
+            // Sort them by priority, uses 0 when no priority was defined
+            List<InjectableBean<?>> priorityDefaultBeans = new ArrayList<>(matching);
+            priorityDefaultBeans.sort(ArcContainerImpl::compareDefaultBeans);
+            Integer highest = priorityDefaultBeans.get(0).getPriority();
+            for (Iterator<InjectableBean<?>> iterator = priorityDefaultBeans.iterator(); iterator.hasNext();) {
+                if (!highest.equals(iterator.next().getPriority())) {
+                    iterator.remove();
+                }
+            }
+            if (priorityDefaultBeans.size() == 1) {
```

Comment:
This entire `if` is useless, no? If `priorityDefaultBeans` has exactly 1 item, then `Set.of(priorityDefaultBeans.get(0))` is exactly the same as `Set.copyOf(priorityDefaultBeans)`. Is this meant as an optimization?

Reply:
OK, apparently the same code is used below too, so let's stay consistent.

Reply:
I have no idea what was the original intention but I kept it this way for the sake of consistency :shrug: